src/main/java/com/redhat/pantheon/sling/PantheonRepositoryInitializer.java

```diff
@@ -41,22 +41,29 @@ public void processRepository(SlingRepository slingRepository) throws Exception
     private void initializeRepositoryACLs(SlingRepository slingRepository) throws RepositoryException {
         JackrabbitSession s = (JackrabbitSession) slingRepository.loginAdministrative(null);
         try {
-            User admin = (User) s.getUserManager().getAuthorizable("admin");
-            admin.changePassword("ccsadmin"); // FIXME - hardcoding admin passwords is a Bad Thing
-
+            // Create and give the pantheon service user permissions to the whole /content path
             try {
                 s.getUserManager().createSystemUser("pantheon", null);
+                s.save();
                 log.info("Created pantheon service account");
-                // Give the pantheon service user permissions to the whole /content path
-                assignPermissionToPrincipal(s, "pantheon", "/content", "*", Privilege.JCR_ALL);
-            } catch (AuthorizableExistsException aee) {
+            } catch (AuthorizableExistsException aeex) {
                 log.info("Pantheon service account already exists");
             }
+            assignPermissionToPrincipal(s, "pantheon", "/content", "*", Privilege.JCR_ALL);
+
+            // Create and grant permissions to the pantheon-users group
+            try {
+                s.getUserManager().createGroup("pantheon-users");
+                s.save();
+                log.info("Created pantheon-users group");
+            } catch (AuthorizableExistsException aeex) {
+                log.info("pantheon-users group already exists");
```

Comment:
I see this is likely the same logic as above - try/catch and s.save() this separately - right?

Reply:
Yes, the only reason to separate them is to be able to log specific entries for each operation attempted. Should give us better insight into what the system is doing.